sky/serve/autoscalers.py

```diff
@@ -171,8 +190,31 @@ def _get_desired_num_replicas(self) -> int:
                 return target_num_replicas
         else:
             self.upscale_counter = self.downscale_counter = 0
+
+        logger.info(
+            f'Final target number of replicas: {self.target_num_replicas} '
+            f'({self.target_num_replicas + self.num_overprovision} with '
+            f'over-provision), Upscale counter: {self.upscale_counter}/'
+            f'{self.scale_up_consecutive_periods}, '
+            f'Downscale counter: {self.downscale_counter}/'
+            f'{self.scale_down_consecutive_periods}')
```

Comment:
moving this log back? It is confusing when the var num_overprovision is not used in this function

Reply:
we can remove `num_overprovision` if it is set to 0. updated.
main reason to put it here is to reduce duplicated code.